source/lib/compiler.ts

```diff
@@ -63,10 +62,12 @@ const ignoreDiagnostic = (
 		return 'preserve';
 	}
 
-	const diagnosticFileName = (diagnostic.file as SourceFile).fileName;
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	const diagnosticFileName = diagnostic.file!.fileName;
```

Comment:
Shouldn't we guard `diagnostic.file`? Or is its optionality incorrect?

Reply:
It doesn't seem to have caused any problems yet. I don't really know whether it may actually be `undefined`.

Reply:
Yeah, it's not really related to this PR. Just thought I'd bring it up. We should be careful about `!` in general unless we are 100% sure it cannot be `undefined` or `null`.